pennylane/templates/layers/strongly_entangling.py

```diff
@@ -92,6 +92,17 @@ def __init__(self, weights, wires, ranges=None, imprimitive=None, do_queue=True)
                 self.ranges = [(l % (len(wires) - 1)) + 1 for l in range(self.n_layers)]
             else:
                 self.ranges = [0] * self.n_layers
+        else:
+            for r in ranges:
+                if r % len(wires) == 0:
+                    raise ValueError(
+                        f"Rangers must not be zero nor divisible by the number of wires; got {r}"
+                    )
+            self.ranges = ranges
+            # repeat range sequence if the length is shorter than the number of layers
+            if len(ranges) < self.n_layers:
+                for i in range(self.n_layers - len(ranges)):
+                    self.ranges.append(self.ranges[i])
```

Comment:
```suggestion
```
I would rather not fill the `ranges` up; if a user decides to fine-tune this hyperparameter the onus is on her to make sure it has the correct size...which also gives better control about what is happening in the template. What do you think?

We could add a check that `len(ranges) == len(self.n_layers)`?

Reply:
Okay, you have me convinced. The user should be responsible for it. The function does look cleaner that way by being simpler and straightforward. I will add a check as you suggested.